MapboxNavigation/CarPlayManager.swift

```diff
@@ -329,18 +329,23 @@ extension CarPlayManager: CPInterfaceControllerDelegate {
         }
     }
     public func templateWillDisappear(_ template: CPTemplate, animated: Bool) {
+        typealias TemplateBooleanClosure = (CPTemplate) -> Bool
+        guard let interface = interfaceController else { return }
+        
+        let listOrSearchOrMap = [CPSearchTemplate.self, CPListTemplate.self, CPMapTemplate.self]
+        let goingToSearch: TemplateBooleanClosure = { type(of: $0) == CPSearchTemplate.self }
+        let currentlyOnHomeScreen = interface.templates.count == 1
+        let presentingFromFavOrSearchOrNav = currentlyOnHomeScreen && listOrSearchOrMap.contains { $0 == type(of: template) }
```

Comment:
This is a bit more indirection than necessary:

```swift
let templateIsForBrowsing = [CPSearchTemplate.self, CPListTemplate.self, CPMapTemplate.self, …].contains(type(of: template))
```

Reply:
`contains(_:)` does not seem to be available for `[CPTemplate.Type]`. I suspect this may be due to the fact that metatypes may not conform to `Equatable`? 🤔